src/test/javascript/spec/component/text-editor/text-editor.spec.ts

```diff
@@ -0,0 +1,166 @@
+                { provide: TextEditorService, useClass: MockTextEditorService },
+                { provide: LocalStorageService, useClass: MockSyncStorage },
+                { provide: SessionStorageService, useClass: MockSyncStorage },
+            ],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(TextEditorComponent);
+                comp = fixture.componentInstance;
+                debugElement = fixture.debugElement;
+                textService = debugElement.injector.get(TextEditorService);
+                getTextForParticipationStub = stub(textService, 'get');
+            });
+    });
+
+    afterEach(() => {
+        getTextForParticipationStub.restore();
+    });
+
+    it('should not allow to submit after the deadline if there is no due date', fakeAsync(() => {
+        const participationSubject = new BehaviorSubject<StudentParticipation>(participation);
+        getTextForParticipationStub.returns(participationSubject);
+        comp.textExercise = textExercise;
+
+        fixture.detectChanges();
+        tick();
+
+        expect(comp.isAllowedToSubmitAfterDeadline).to.be.false;
+    }));
+
+    it('should not allow to submit after the deadline if the initialization date is before the due date', fakeAsync(() => {
+        participation.initializationDate = moment();
+        textExercise.dueDate = moment().add(1, 'days');
+        const participationSubject = new BehaviorSubject<StudentParticipation>(participation);
+        getTextForParticipationStub.returns(participationSubject);
+        comp.textExercise = textExercise;
+
+        fixture.detectChanges();
+        tick();
+
+        expect(comp.isAllowedToSubmitAfterDeadline).to.be.false;
+    }));
+
+    it('should allow to submit after the deadline if the initilization date is after the due date', fakeAsync(() => {
+        participation.initializationDate = moment().add(1, 'days');
+        textExercise.dueDate = moment();
+        const participationSubject = new BehaviorSubject<StudentParticipation>(participation);
+        getTextForParticipationStub.returns(participationSubject);
+        comp.textExercise = textExercise;
+
+        fixture.detectChanges();
+        tick();
+
+        expect(comp.isAllowedToSubmitAfterDeadline).to.be.true;
+    }));
+
+    it('should not be always active if there is a result and no due date', fakeAsync(() => {
+        const participationSubject = new BehaviorSubject<StudentParticipation>(participation);
+        getTextForParticipationStub.returns(participationSubject);
+        comp.result = result;
+        comp.textExercise = textExercise;
+
+        fixture.detectChanges();
+        tick();
+
+        expect(comp.isAlwaysActive).to.be.false;
+    }));
+
+    it('should be always active if there is no result and no due date', fakeAsync(() => {
+        const participationSubject = new BehaviorSubject<StudentParticipation>(participation);
```

Comment:
You already had this setup in the first test. I think it makes sense to reuse it and check both values